vault/src/system.rs

```diff
@@ -0,0 +1,412 @@
+                bitcoin_core.clone(),
+                issue_set.clone(),
+                num_confirmations,
+            ),
+        );
+
+        // replace handling
+        let (replace_event_tx, replace_event_rx) = mpsc::channel::<RequestEvent>(16);
+        let (replace_block_tx, replace_block_rx) = mpsc::channel::<PolkaBtcHeader>(16);
+
+        let request_replace_listener = wait_or_shutdown(
+            self.shutdown.clone(),
+            listen_for_replace_requests(
+                self.btc_parachain.clone(),
+                bitcoin_core.clone(),
+                replace_event_tx.clone(),
+                !self.config.no_auto_replace,
+            ),
+        );
+
+        let accept_replace_listener = wait_or_shutdown(
+            self.shutdown.clone(),
+            listen_for_accept_replace(
+                self.btc_parachain.clone(),
+                bitcoin_core.clone(),
+                num_confirmations,
+            ),
+        );
+
+        let execute_replace_listener = wait_or_shutdown(
+            self.shutdown.clone(),
+            listen_for_execute_replace(self.btc_parachain.clone(), replace_event_tx.clone()),
+        );
+
+        let auction_replace_listener = wait_or_shutdown(
+            self.shutdown.clone(),
+            listen_for_auction_replace(
+                self.btc_parachain.clone(),
+                bitcoin_core.clone(),
+                num_confirmations,
+            ),
+        );
+
+        let mut replace_cancellation_scheduler =
+            CancellationScheduler::new(self.btc_parachain.clone(), vault_id.clone());
+
+        let replace_block_provider = self.btc_parachain.clone();
+        let replace_block_listener = wait_or_shutdown(self.shutdown.clone(), async move {
+            let replace_block_tx = &replace_block_tx;
+            replace_block_provider
+                .on_block(move |header| async move {
+                    replace_block_tx
+                        .clone()
+                        .send(header.clone())
+                        .await
+                        .map_err(|_| RuntimeError::ChannelClosed)?;
+                    Ok(())
+                })
+                .await
+        });
+
+        let replace_cancel_scheduler = wait_or_shutdown(self.shutdown.clone(), async move {
+            replace_cancellation_scheduler
+                .handle_cancellation::<ReplaceCanceller>(replace_block_rx, replace_event_rx)
+                .await
+        });
+
+        let third_party_collateral_listener = wait_or_shutdown(
+            self.shutdown.clone(),
+            monitor_collateral_of_vaults(
+                self.btc_parachain.clone(),
+                bitcoin_core.clone(),
+                replace_event_tx.clone(),
+                self.config.collateral_timeout,
+            ),
+        );
+
+        // redeem handling
+        let redeem_listener = wait_or_shutdown(
+            self.shutdown.clone(),
+            listen_for_redeem_requests(
```

Comment:
This function (as well as some other functions) internally spawns new tasks that will not be cancelled when the shutdown signal is called

Reply:
I think that should be okay for now, though I may re-write these later to receive the shutdown signal.